src/telemetry-data/data-model.md

```diff
@@ -0,0 +1,203 @@
+  * but are not *actually* thin-edge JSON,
+  * making mandatory yet another data mapper.
+* Thin-edge JSON is focused on a format, while what we need first is a data model.
+* The current format parser is too rigid,
+  * rejecting the whole message when only one part is not meaningful.
+* Many data sources are publishing data with one topic per kind of data
+  * while thin-edge expect all the data on a single topic.
+* Writing a mapper that groups messages as those produced by `collectd` is a more complex task than expected.
+  * One should leverage the effort done for `collectd` and use it 
+* All the data published on `tedge/measurements` are sent to the cloud.
+  * What if we wish to filter some data out?
+  * What if we wish to pre-process the data to only forward an aggregate?
+* The meaning and purpose of multi-measurements is not clear.
+* Possible name collisions with reserved key words (`time` and `type`)?
+* Non numeric values are rejected.
+* Accept only ISO-8601 timestamp, while Unix Epoch is often used by data sources. 
+* The child device identifiers are expected to be in the topic name,
+  * while some data sources published their identifier along the message.
+* There is no way to attach metadata to a data source (e.g the unit).
+* There is no way to attach metadata to a data point (e.g the quality of the measurement).
+
+## Requirements for thin-edge JSON version 1.0
+
+* Provide a clear semantics of the data that can be moved around thin-edge components.
+  * Given two sets of messages, are they equivalent or not?
+* Provide a reference mapper that move data around topics from different producers and consumers.
+* Prefer convention to configuration
+  * By convention, messages are exchanged over pre-defined topics - with a format defined by the topic.
+  * Configuration is required only to overcome a convention.
+    Ingesting data from `collectd`, should be as easy as telling thin-edge
+    to ingest not only `tedge/measurements` but also `collectd/raspberrypi`.
+* Ingest data from the main device as well as from child devices / named components.
+* Provide clear equivalence between messages independently of their shape. For instance, between:
+  * `{ "memory": { "used": 1085927424 } }` published on `tedge/measurements`
+  * `{ "memory/used": 1085927424 }` published on `tedge/measurements`
+  * `1085927424` published on `some/input/memory/used`
+* Accept ISO-8601 and Unix Epoch timestamps
+* Provide time-window batching of messages
+* Take thin-edge as a reference to provide simple mapping configuration.
+  * e.g: `[collectd/raspberrypi/${measurement.group}/${measurement.name}] ${measurement.epoch}:${measurement.value}`
+* Let the data sources add any meta-data even if not processed by thin-edge.
+  * It must be okay for a sensor to provide a serial number along its measurements.
+  * It must be okay for an OPCUA data source to tag the measurements with a quality code. 
+
```

Comment:
Components should have the possibility to only listen to a subset of measurements/events

```suggestion
Provide possibility to subscribe to specific measurements/alarms/events
```